From c32ad4876d4936e256b7daf7d9e0f44545593212 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Tue, 15 Jul 2025 08:10:19 +0000
Subject: [PATCH 1/2] LOCAL: CEGLFence: Ignore improper atomic drmrequest when
 async rendering and

This reverts commit 54b9ccdb4fd8c6397846cd35887292dbb31cf3b1.

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 xbmc/utils/EGLFence.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/xbmc/utils/EGLFence.cpp b/xbmc/utils/EGLFence.cpp
index 58a9ec1683..9d0065bdaf 100644
--- a/xbmc/utils/EGLFence.cpp
+++ b/xbmc/utils/EGLFence.cpp
@@ -133,7 +133,11 @@ void CEGLFence::WaitSyncCPU()
   if (!m_kmsFence)
     return;
 
-  if (m_eglClientWaitSyncKHR(m_display, m_kmsFence, 0, EGL_FOREVER_KHR) != EGL_FALSE)
-    m_eglDestroySyncKHR(m_display, m_kmsFence);
+  EGLint status{EGL_FALSE};
+
+  while (status != EGL_CONDITION_SATISFIED_KHR)
+    status = m_eglClientWaitSyncKHR(m_display, m_kmsFence, 0, EGL_FOREVER_KHR);
+
+  m_eglDestroySyncKHR(m_display, m_kmsFence);
 }
 #endif
-- 
2.34.1

